### ISSUE TITLE
TF version warnings 

### ISSUE THREAD

**Issue by IyadL** [knowledgedefinednetworking/RouteNet-challenge#4]
Hi 
I installed the TF version of the challenge. It is running but I got some warnings: 
`C:\Users\ia-te_5pzizb8\AppData\Local\Programs\Python\Python37\lib\site-packages\tensorflow\python\framework\indexed_slices.py:437: UserWarning: Converting sparse IndexedSlices(IndexedSlices(indices=Tensor("gradients/GatherV2_1_grad/Reshape_1:0", shape=(None,), dtype=int32), values=Tensor("gradients/GatherV2_1_grad/Reshape:0", shape=(None, 32), dtype=float32), dense_shape=Tensor("gradients/GatherV2_1_grad/Cast:0", shape=(2,), dtype=int32))) to a dense Tensor of unknown shape. This may consume a large amount of memory.                                                                                                    "shape. This may consume a large amount of memory." % value)  `

I tried some solutions from the internet but didn't work for me. 
https://stackoverflow.com/questions/35892412/tensorflow-dense-gradient-explanation

Any recommendations to solve this problem please ? 

I am using: 
OS: windows 10 
TF version: 2.4.1 
CUDA version: 11.2
Python version: 3.7.0

**Comment by MiquelFerriol** [comment]
Hi @moby91,
First of all, thank you for your interest in the GNNet Challenge.

Secondly, the warning you are describing is known and does not really affect the execution. This Warning is raised by the tf.gather function that needs to convert the tf.IndexedSlices to a dense tensor. The problem is that, since RouteNet is working with graph-structured data, we do not really know the shape of the tensors the model gets as input. We already tried some solutions like the one you mentioned but, either didn't work or they have an important effect on the performance. So that's why we decided to keep this implementation even if it rises those warnings (note that you would get one warning for each message passing iteration the model executes).

Anyway, you should be able to execute and train the model quite efficiently since the graphs aren't very large. This could possibly have some impact on the evaluation dataset since graphs are considerably larger (however, 16GB of memory should be enough to handle them).

Bests,
Miquel
